src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -598,8 +623,8 @@ public static void setCacheSize(int cacheSize) {
     /** {@inheritDoc} */
     @Override
     public String getRemote() {
-        return GitHubSCMBuilder.uriResolver(getOwner(), apiUri, credentialsId)
-                .getRepositoryUri(apiUri, repoOwner, repository);
+        // Only HTTPS is applicable to the source remote with Username / Password credentials
```

Comment:
I don't think you can be sure it is only username / password credentials?

Reply:
It could be an app credential couldn't it?

Reply:
> It could be an app credential couldn't it?

Yes it is, but `GitHubAppCredentials` implements `StandardUsernamePasswordCredentials` and is used for REST calls over HTTPS as well as mentioned Alan. So "HTTPS resolver only" here looks good to me.